app/services/recovery_code_generator.rb

```diff
@@ -7,16 +7,16 @@ def initialize(user, length: 16)
     @key_maker = EncryptedKeyMaker.new
   end
 
-  def create
-    @user_access_key = make_user_access_key(raw_recovery_code)
+  def create(salt = user.password_salt)
```

Comment:
Seeing as it's been easy for us to forget which salt goes where, what do you think about making this a required param so it's more explicit which salt we use each time and where?